Sources/CMakeLists.txt

```
@@ -6,28 +6,106 @@
+    set(SwiftTesting_BuildMacrosAsExecutables NO)
+  else()
+    set(SwiftTesting_BuildMacrosAsExecutables YES)
+  endif()
 
-find_package(SwiftSyntax CONFIG GLOBAL)
-if(SwiftSyntax_FOUND)
-  set(SwiftTesting_BuildMacrosAsExecutables NO)
+  # Build and install the plugin into the current build directry.
+  set(SwiftTesting_MACRO_INSTALL_PREFIX "${CMAKE_BINARY_DIR}/plugin")
+
+  ExternalProject_Add(TestingMacros
+    PREFIX "tm"
+    SOURCE_DIR "${CMAKE_CURRENT_SOURCE_DIR}/TestingMacros"
+    BUILD_ALWAYS ON
+    CMAKE_ARGS
+      -DCMAKE_MAKE_PROGRAM=${SwiftTesting_MACRO_MAKE_PROGRAM}
+      -DCMAKE_Swift_COMPILER=${SwiftTesting_MACRO_Swift_COMPILER}
+      -DCMAKE_Swift_FLAGS=${SwiftTesting_MACRO_Swift_FLAGS}
+      -DCMAKE_Swift_FLAGS_RELEASE=${SwiftTesting_MACRO_Swift_FLAGS_RELEASE}
+      -DCMAKE_Swift_FLAGS_RELWITHDEBINFO=${SwiftTesting_MACRO_Swift_FLAGS_RELWITHDEBINFO}
+      -DCMAKE_AR=${SwiftTesting_MACRO_AR}
+      -DCMAKE_RANLIB=${SwiftTesting_MACRO_RANLIB}
+      -DCMAKE_BUILD_TYPE=${CSwiftTesting_MACRO_BUILD_TYPE}
+      -DSwiftTesting_BuildMacrosAsExecutables=${SwiftTesting_BuildMacrosAsExecutables}
+      -DSwiftSyntax_DIR=${SwiftSyntax_DIR}
+      -DCMAKE_INSTALL_PREFIX=${SwiftTesting_MACRO_INSTALL_PREFIX})
+
+  # Hardcode the known file names based on system name as a workaround since
+  # TestingMacros uses `ExternalProject` and we cannot directly query the
+  # properties of its targets here.
+  if(NOT SwiftTesting_BuildMacrosAsExecutables)
+    if(CMAKE_HOST_SYSTEM_NAME STREQUAL "Darwin")
+      set(SwiftTesting_MACRO_PATH "${SwiftTesting_MACRO_INSTALL_PREFIX}/lib/libTestingMacros.dylib")
+      install(PROGRAMS "${SwiftTesting_MACRO_PATH}"
+        DESTINATION lib/swift/host/plugins/testing)
+    elseif(CMAKE_HOST_SYSTEM_NAME STREQUAL "Linux")
+      set(SwiftTesting_MACRO_PATH "${SwiftTesting_MACRO_INSTALL_PREFIX}/lib/libTestingMacros.so")
+      install(PROGRAMS "${SwiftTesting_MACRO_PATH}"
+        DESTINATION lib/swift/host/plugins)
+    elseif(CMAKE_HOST_SYSTEM_NAME STREQUAL "Windows")
+      set(SwiftTesting_MACRO_PATH "${SwiftTesting_MACRO_INSTALL_PREFIX}/bin/TestingMacros.dll")
+      install(PROGRAMS "${SwiftTesting_MACRO_PATH}"
+        DESTINATION bin)
+    else()
+      message(FATAL_ERROR "Unable to determine the library name for TestingMacros based on system name: ${CMAKE_HOST_SYSTEM_NAME}")
+    endif()
+  else()
+    if(CMAKE_HOST_SYSTEM_NAME STREQUAL "Windows")
+      set(SwiftTesting_MACRO_PATH "${SwiftTesting_MACRO_INSTALL_PREFIX}/bin/TestingMacros.exe")
+    else()
+      set(SwiftTesting_MACRO_PATH "${SwiftTesting_MACRO_INSTALL_PREFIX}/bin/TestingMacros")
+    endif()
+  endif()
+elseif(SwiftTesting_MACRO)
+  # Use the passed-in plugin path.
+  set(SwiftTesting_MACRO_PATH "${SwiftTesting_MACRO}")
+  add_custom_target(TestingMacros DEPENDS "${SwiftTesting_MACRO_PATH}")
 else()
-  set(SwiftTesting_BuildMacrosAsExecutables YES)
+  # If it's excplicitly "NO", do not compile the libray with macros.
```

Minor typos that can be fixed if there is need to iterate on other changes.

```suggestion
  # If it's explicitly "NO", do not compile the library with macros.
```